---
title: "250205_snRNAseq_QC_clean"
author: "Simona Sarafinovska"
date: "2025-02-05"
---

# Set up environment
We first need to tell RStudio where to find our packages. 

```{r dependencies, warning=FALSE, message=FALSE}
# Change this to your directory
.libPaths('/ref/jdlab/software/r-envs/simona/r-packages/4.2/')

library(Seurat)
library(scCustomize)
library(DoubletFinder)
library(ggplot2)
library(patchwork)
library(tidyverse)

source("../color_utils.R")

set.seed(420)
```

# Load in data
```{r load combined object}
combined <- readRDS("../../data/250219_combined_1-5_5pct.rds")
```

# Perform final merging of the objects
```{r final merging}
combined$genotype <- factor(combined$genotype, levels=c('WT', 'Het'))
combined$sample <- factor(combined$sample, levels=c("Het_F_1",
"Het_F_2",
"Het_F_3",
"Het_F_4",
"Het_F_5",
"Het_F_6",
"WT_F_1",
"WT_F_2",
"WT_F_3",
"WT_F_4",
"WT_F_5",
"WT_F_6",
"WT_F_7",
"WT_F_8",
"WT_F_9",
"WT_F_10",
"Het_M_1",
"Het_M_2",
"Het_M_3",
"Het_M_4",
"Het_M_5",
"Het_M_6",
"Het_M_7",
"Het_M_8",
"Het_M_9",
"Het_M_10",
"WT_M_1",
"WT_M_2",
"WT_M_3",
"WT_M_4",
"WT_M_5",
"WT_M_6",
"WT_M_7",
"WT_M_8",
"WT_M_9",
"WT_M_10"))

combined
```

You now have a Seurat Object that contains all the nuclei. A quick look the object shows that there are 37462 features (genes) and 184274 samples (nuclei). In the following quality control steps, we will be filtering out low quality nuclei and potential doublets. 

# QC and filtering

## Pre-processing and quality control

Quality control is an important step to make sure that the data is appropriately filtered. There are numerous ways to perform QC, here we will complement the above metrics by plotting and checking visually. This is purely for visualization purposes for now, especially given prior filtering during loading step.

We can plot the number of genes per nucleus for each sample. 
```{r plot genes per nuc}
QC_Plots_Genes(seurat_object = combined, 
               group.by = 'sample',
               low_cutoff = 300, 
               high_cutoff = 7000, 
               y_axis_log = TRUE,
               plot_title = 'Genes per nucleus')

ggsave('../../results/processing/intermed-figs/250221_combined_genepernuc.png',
       device = png,
       width = 6,
       height = 3)
```

We can plot the number of UMIs per nucleus for each sample. The specified cutoff values are arbitrary now and can be changed. This is purely for visualization purposes for now.

```{r plot UMIs per nuc}
QC_Plots_UMIs(seurat_object = combined, 
              group.by = 'sample',
              low_cutoff = 300, 
              high_cutoff = 40000,
              y_axis_log = FALSE,
              plot_title = 'UMIs per nucleus')

ggsave('../../results/processing/intermed-figs/250221_combined_UMIspernuc.png',
       device = png,
       width = 6,
       height = 3)
```

We can plot the percentage of mitochondrial genes per nucleus for each sample. The specified cutoff values are arbitrary now and can be changed. This is purely for visualization purposes for now.

```{r plot pct mito}
QC_Plots_Mito(seurat_object = combined, 
              group.by = 'sample',
              high_cutoff = 5,
              plot_title = 'Mito Gene % per nucleus')

ggsave('../../results/processing/intermed-figs/250221_combined_mitopernuc.png',
       device = png,
       width = 6,
       height = 3)
```

We can plot the complexity score for each sample. The specified cutoff values are arbitrary now and can be changed. This is purely for visualization purposes for now.

```{r plot complexity score}
QC_Plots_Complexity(seurat_object = combined, 
                    group.by = 'sample',
                    high_cutoff = 0.85, 
                    plot_title = 'Complexity score per nucleus')

ggsave('../../results/processing/intermed-figs/250221_combined_complexitypernuc.png',
       device = png,
       width = 6,
       height = 3)
```

```{r plot UMI v gene}
QC_Plot_UMIvsGene(seurat_object = combined, 
                  group.by = 'sample',
                  low_cutoff_gene = 300, high_cutoff_gene = 7000,
                  low_cutoff_UMI = 800, high_cutoff_UMI = 40000,
                  )

ggsave('../../results/processing/intermed-figs/250221_combined_UMIvsgene.png',
       device = png,
       width = 6,
       height = 3)
```

```{r plot gene per feature}
QC_Plot_GenevsFeature(seurat_object = combined,
                      group.by = 'sample',
                      feature1 = "percent_mito",
                      low_cutoff_gene = 300, high_cutoff_gene = 7000,
                      high_cutoff_feature = 5)

ggsave('../../results/processing/intermed-figs/250221_combined_genesperfeature.png',
       device = png,
       width = 6,
       height = 3)
```

We can calculate some basic QC stats and return a dataframe.

```{r median table}
median_stats <- Median_Stats(seurat_object = combined, group_by_var = 'sample')
median_stats
```

We can now plot the median values.

```{r plot medians}
p1 <- Plot_Median_Genes(seurat_object = combined, sample_col = 'sample', group_by = 'genotype') + 
  scale_fill_manual(values = palette_geno) +
  ggtitle('Median Genes') +
  theme(legend.position = 'none')
p2 <- Plot_Median_UMIs(seurat_object = combined, sample_col = 'sample', group_by = 'genotype') + 
  scale_fill_manual(values = palette_geno) +
  ggtitle('Median UMIs') +
  theme(legend.position = 'none')
wrap_plots(p1, p2, ncol = 2)

ggsave('../../results/processing/intermed-figs/250221_combined_medians.png',
       device = png,
       width = 6,
       height = 3)
```

## Removal of doublets and low quality cells

```{r removal doublets}
combined <- subset(combined, subset = doublet == "Singlet")
combined
```

You should now have an object with 37462 features across 176009 nuclei. 

I am changing my previously used min UMI threshold (nCount_RNA) here to 800 (from a less stringent 400).

```{r changing UMI threshold}
combined <- subset(combined, subset = nCount_RNA > 800)
combined
```

You should now have an object with 37462 features across 138025 nuclei. 

# QC plots

QC metrics and library statistics after filtering 

```{r genes/umis per sample}
p3 <- QC_Plots_Genes(seurat_object = combined, 
                     group.by = 'sample',
                     pt.size = 0)
p4 <- QC_Plots_UMIs(seurat_object = combined,
                    group.by = 'sample',
                    pt.size = 0)

wrap_plots(p3, p4, ncol = 2)
```

```{r nucs per sample}
Plot_Cells_per_Sample(seurat_object = combined, sample_col = 'sample', group_by = 'genotype')
```

This shows the number of nuclei per sample.

```{r table nucs per sample}
samplesTable <- table(combined@meta.data$sample)
samplesTable
```

This shows the number of nuclei per genotype.

```{r table nucs per geno}
genotypesTable <- table(combined@meta.data$genotype)
genotypesTable
```

This creates a dataframe that summarizes the statistics across all samples and genotypes.

```{r df stats geno}
df <- data.frame(
  numCells = as.vector(samplesTable),
  genotype = c(rep(c("Het"), times = 16), rep(c("WT"), times = 20))
)

df$genotype <- factor(df$genotype, levels = c("WT", "Het"))

averages <- aggregate(numCells ~ genotype, data = df, FUN = mean)
sds <-aggregate(numCells ~ genotype, data = df, FUN = function(x) sd(x))

df2 <- merge(averages, sds, by = "genotype")
colnames(df2) <- c("genotype", "mean", "sd")

df2$genotype <- factor(df2$genotype, levels = c("WT", "Het"))
df2 <- df2[order(df2$genotype), ]

df 
df2
```

We will now plot 3 plots. 
1. A barplot that shows the average number of nuclei per replicate per genotype.
2. A boxplot that shows the distribution of genes per nuclei per replicate.
3. A boxplot that shows the distribution of UMIs per nuclei per replicate.
```{r bar and boxplots geno}
p1 <- ggplot(df2, aes(x = genotype, y = mean)) +
  geom_bar(stat = "identity",
           fill = palette_geno) +
  geom_errorbar(aes(x = genotype, ymin = mean-sd, ymax = mean+sd),
                width = 0.2,
                linewidth = 0.2) +
  geom_jitter(data = df, aes(y = numCells, x = genotype),
              width = 0.2, size = 1) +
  xlab('') +
  ylab('') +
  ggtitle('Nuclei') +
  theme_classic() +
  theme(plot.title = element_text(size = 14),
        axis.text.y = element_text(size = 8))

p2 <- ggplot(combined@meta.data, aes(x = genotype, y = nFeature_RNA, fill = sample)) +
  geom_boxplot(outlier.color = NA,
               linewidth = 0.2,
               varwidth = TRUE) +
  scale_fill_manual(values = c(WT_F_1 = '#eecc67', 
                  WT_F_2 = '#eecc67', 
                  WT_F_3 = '#eecc67',
                  WT_F_4 = '#eecc67', 
                  WT_F_5 = '#eecc67', 
                  WT_F_6 = '#eecc67',
                  WT_F_7 = '#eecc67',
                  WT_F_8 = '#eecc67', 
                  WT_F_9 = '#eecc67', 
                  WT_F_10 = '#eecc67',
                  Het_F_1 = '#98782d', 
                  Het_F_2 = '#98782d', 
                  Het_F_3 = '#98782d',
                  Het_F_4 = '#98782d', 
                  Het_F_5 = '#98782d', 
                  Het_F_6 = '#98782d',
                  WT_M_1 = '#eecc67', 
                  WT_M_2 = '#eecc67', 
                  WT_M_3 = '#eecc67',
                  WT_M_4 = '#eecc67', 
                  WT_M_5 = '#eecc67', 
                  WT_M_6 = '#eecc67',
                  WT_M_7 = '#eecc67',
                  WT_M_8 = '#eecc67', 
                  WT_M_9 = '#eecc67', 
                  WT_M_10 = '#eecc67',
                  Het_M_1 = '#98782d', 
                  Het_M_2 = '#98782d', 
                  Het_M_3 = '#98782d',
                  Het_M_4 = '#98782d', 
                  Het_M_5 = '#98782d', 
                  Het_M_6 = '#98782d',
                  Het_M_7 = '#98782d',
                  Het_M_8 = '#98782d', 
                  Het_M_9 = '#98782d', 
                  Het_M_10 = '#98782d')) +
  xlab('') +
  ylab('') +
  ggtitle('Genes') +
  ylim(0, 4000) +
  theme_classic() +
  theme(legend.position = "none",
        plot.title = element_text(size = 14),
        axis.text.y = element_text(size = 8))

p3 <- ggplot(combined@meta.data, aes(x = genotype, y = nCount_RNA, fill = sample)) +
  geom_boxplot(outlier.color = NA,
               linewidth = 0.2,
               varwidth = TRUE) +
  scale_fill_manual(values = c(WT_F_1 = '#eecc67', 
                  WT_F_2 = '#eecc67', 
                  WT_F_3 = '#eecc67',
                  WT_F_4 = '#eecc67', 
                  WT_F_5 = '#eecc67', 
                  WT_F_6 = '#eecc67',
                  WT_F_7 = '#eecc67',
                  WT_F_8 = '#eecc67', 
                  WT_F_9 = '#eecc67', 
                  WT_F_10 = '#eecc67',
                  Het_F_1 = '#98782d', 
                  Het_F_2 = '#98782d', 
                  Het_F_3 = '#98782d',
                  Het_F_4 = '#98782d', 
                  Het_F_5 = '#98782d', 
                  Het_F_6 = '#98782d',
                  WT_M_1 = '#eecc67', 
                  WT_M_2 = '#eecc67', 
                  WT_M_3 = '#eecc67',
                  WT_M_4 = '#eecc67', 
                  WT_M_5 = '#eecc67', 
                  WT_M_6 = '#eecc67',
                  WT_M_7 = '#eecc67',
                  WT_M_8 = '#eecc67', 
                  WT_M_9 = '#eecc67', 
                  WT_M_10 = '#eecc67',
                  Het_M_1 = '#98782d', 
                  Het_M_2 = '#98782d', 
                  Het_M_3 = '#98782d',
                  Het_M_4 = '#98782d', 
                  Het_M_5 = '#98782d', 
                  Het_M_6 = '#98782d',
                  Het_M_7 = '#98782d',
                  Het_M_8 = '#98782d', 
                  Het_M_9 = '#98782d', 
                  Het_M_10 = '#98782d')) +
  xlab('') +
  ylab('') +
  ggtitle('UMIs') +
  ylim(0, 7500) +
  theme_classic() +
  theme(legend.position = "none",
        plot.title = element_text(size = 14),
        axis.text.y = element_text(size = 8))

wrap_plots(p1, p2, p3, ncol = 3)

ggsave('../../results/processing/final-figs/250221_combined_1-5_5pct_genoQC.pdf',
       device = pdf,
       width = 9,
       height = 3)
```

Same but for Sex.
```{r add sex to metadata}
meta <- combined@meta.data$sample
new_meta <- gsub(".*_([^_]+)_.*", "\\1", levels(meta))
levels(meta) <- new_meta

combined <- AddMetaData(combined, meta, col.name = 'sex')
```

```{r df stats sex}
df <- data.frame(
  numCells = as.vector(samplesTable),
  sex = c(rep(c("F"), times = 6), rep(c("M"), times = 10),rep(c("F"), times = 10), rep(c("M"), times = 10))
)

df$sex <- factor(df$sex, levels = c("M", "F"))

averages <- aggregate(numCells ~ sex, data = df, FUN = mean)
sds <-aggregate(numCells ~ sex, data = df, FUN = function(x) sd(x))

df2 <- merge(averages, sds, by = "sex")
colnames(df2) <- c("sex", "mean", "sd")

df2$sex <- factor(df2$sex, levels = c("M", "F"))
df2 <- df2[order(df2$sex), ]

df 
df2
```

```{r bar and boxplots sex}
p1 <- ggplot(df2, aes(x = sex, y = mean)) +
  geom_bar(stat = "identity",
           fill = palette_sex) +
  geom_errorbar(aes(x = sex, ymin = mean-sd, ymax = mean+sd),
                width = 0.2,
                linewidth = 0.2) +
  geom_jitter(data = df, aes(y = numCells, x = sex),
              width = 0.2, size = 1) +
  xlab('') +
  ylab('') +
  ggtitle('Nuclei') +
  theme_classic() +
  theme(plot.title = element_text(size = 14),
        axis.text.y = element_text(size = 8))

p2 <- ggplot(combined@meta.data, aes(x = sex, y = nFeature_RNA, fill = sample)) +
  geom_boxplot(outlier.color = NA,
               linewidth = 0.2,
               varwidth = TRUE) +
  scale_fill_manual(values = c(WT_F_1 = '#ED6677', 
                  WT_F_2 = '#ED6677', 
                  WT_F_3 = '#ED6677',
                  WT_F_4 = '#ED6677', 
                  WT_F_5 = '#ED6677', 
                  WT_F_6 = '#ED6677',
                  WT_F_7 = '#ED6677',
                  WT_F_8 = '#ED6677', 
                  WT_F_9 = '#ED6677', 
                  WT_F_10 = '#ED6677',
                  Het_F_1 = '#ED6677', 
                  Het_F_2 = '#ED6677', 
                  Het_F_3 = '#ED6677',
                  Het_F_4 = '#ED6677', 
                  Het_F_5 = '#ED6677', 
                  Het_F_6 = '#ED6677', 
                  WT_M_1 = '#4478AB', 
                  WT_M_2 = '#4478AB', 
                  WT_M_3 = '#4478AB',
                  WT_M_4 = '#4478AB', 
                  WT_M_5 = '#4478AB', 
                  WT_M_6 = '#4478AB',
                  WT_M_7 = '#4478AB',
                  WT_M_8 = '#4478AB', 
                  WT_M_9 = '#4478AB', 
                  WT_M_10 = '#4478AB',
                  Het_M_1 = '#4478AB', 
                  Het_M_2 = '#4478AB', 
                  Het_M_3 = '#4478AB',
                  Het_M_4 = '#4478AB', 
                  Het_M_5 = '#4478AB', 
                  Het_M_6 = '#4478AB',
                  Het_M_7 = '#4478AB',
                  Het_M_8 = '#4478AB', 
                  Het_M_9 = '#4478AB', 
                  Het_M_10 = '#4478AB')) +
  xlab('') +
  ylab('') +
  ggtitle('Genes') +
  ylim(0, 4000) +
  theme_classic() +
  theme(legend.position = "none",
        plot.title = element_text(size = 14),
        axis.text.y = element_text(size = 8))


p3 <- ggplot(combined@meta.data, aes(x = sex, y = nCount_RNA, fill = sample)) +
  geom_boxplot(outlier.color = NA,
               linewidth = 0.2,
               varwidth = TRUE) +
  scale_fill_manual(values = c(WT_F_1 = '#ED6677', 
                  WT_F_2 = '#ED6677', 
                  WT_F_3 = '#ED6677',
                  WT_F_4 = '#ED6677', 
                  WT_F_5 = '#ED6677', 
                  WT_F_6 = '#ED6677',
                  WT_F_7 = '#ED6677',
                  WT_F_8 = '#ED6677', 
                  WT_F_9 = '#ED6677', 
                  WT_F_10 = '#ED6677',
                  Het_F_1 = '#ED6677', 
                  Het_F_2 = '#ED6677', 
                  Het_F_3 = '#ED6677',
                  Het_F_4 = '#ED6677', 
                  Het_F_5 = '#ED6677', 
                  Het_F_6 = '#ED6677', 
                  WT_M_1 = '#4478AB', 
                  WT_M_2 = '#4478AB', 
                  WT_M_3 = '#4478AB',
                  WT_M_4 = '#4478AB', 
                  WT_M_5 = '#4478AB', 
                  WT_M_6 = '#4478AB',
                  WT_M_7 = '#4478AB',
                  WT_M_8 = '#4478AB', 
                  WT_M_9 = '#4478AB', 
                  WT_M_10 = '#4478AB',
                  Het_M_1 = '#4478AB', 
                  Het_M_2 = '#4478AB', 
                  Het_M_3 = '#4478AB',
                  Het_M_4 = '#4478AB', 
                  Het_M_5 = '#4478AB', 
                  Het_M_6 = '#4478AB',
                  Het_M_7 = '#4478AB',
                  Het_M_8 = '#4478AB', 
                  Het_M_9 = '#4478AB', 
                  Het_M_10 = '#4478AB')) +
  xlab('') +
  ylab('') +
  ggtitle('UMIs') +
  ylim(0, 7500) +
  theme_classic() +
  theme(legend.position = "none",
        plot.title = element_text(size = 14),
        axis.text.y = element_text(size = 8))

wrap_plots(p1, p2, p3, ncol = 3)

ggsave('../../results/processing/final-figs/250221_combined_1-5_5pct_sexQC.pdf',
       device = pdf,
       width = 9,
       height = 3)
```

# Save 

```{r save data}
saveRDS(combined, "../../data/250221_combined_postQC_1-5_5pct.rds")
```
