---
title: "Differential gene expression analysis of MYT1L tuberal hypothalamus after social operant conditioning"
author: "Simona Sarafinovska"
date: "2025-2-18"
---

# Load dependencies

```{r dependencies, warning=FALSE, message=FALSE}

# Change this to your directory
.libPaths('/ref/jdlab/software/r-envs/simona/r-packages/4.2/')

library(Seurat)
library(tidyverse)
library(magrittr)
library(ReactomePA)
library(clusterProfiler)
library(enrichplot)
library(org.Mm.eg.db)
library(DOSE)
library(cowplot)
library(Matrix.utils)
library(edgeR)
library(Matrix)
library(reshape2)
library(S4Vectors)
library(SingleCellExperiment)
library(pheatmap)
library(apeglm)
library(png)
library(DESeq2)
library(RColorBrewer)
library(data.table)
library(viridis)
library(ggrepel)

source("/scratch/jdlab/simona/color_utils.R")
```

# Load in pre-processed data

```{r load rds}
df <-  readRDS("/scratch/jdlab/simona/socop-hyp-snRNAseq/data/250228_combined.neuronal_1-5_5pct.rds")
```

```{r name_cleaning}
df@meta.data$sample <- as.factor(gsub("_", "-", df@meta.data$sample))
df@meta.data$cell_type <- Idents(df)
df@meta.data$cell_type <- as.factor(gsub("/", "-", df@meta.data$cell_type))
df@meta.data$cell_type <- as.factor(gsub(" ", "-", df@meta.data$cell_type))
df@meta.data$cell_type <- as.factor(gsub("_", "-", df@meta.data$cell_type))
```


```{r geno table}
genotypesTable <- table(df@meta.data$genotype)
genotypesTable
```

```{r sex table}
sexesTable <- table(df@meta.data$sex)
sexesTable
```

```{r make sce df}
counts <- df@assays$RNA@counts
metadata <- df@meta.data
sce <- SingleCellExperiment(assay = list(counts = counts),
                            colData = metadata)
```

# Preparing the dataset for pseudobulk analysis

Make sure ALL / and _ are replaced with - .
```{r confirm cluster names cleaned}
cluster_names <- levels(colData(sce)$cell_type)
cluster_names
```

```{r confirm sample names cleaned}
sample_names <- levels(factor(colData(sce)$sample))
sample_names
```

```{r groups df}
groups <- colData(sce)[, c("cell_type", "sample")]
head(groups)
```

```{r aggregate count mtx}
aggr_counts <- aggregate.Matrix(t(counts(sce)),
                                groupings = groups,
                                fun = "sum")
aggr_counts <- t(aggr_counts)
tstrsplit(colnames(aggr_counts), "_") %>% str()
```

```{r aggr count matrix by cluster}
counts_ls <- list()

for (i in 1:length(cluster_names)) {
  column_idx <- which(tstrsplit(colnames(aggr_counts), "_")[[1]] == cluster_names[i])
  counts_ls[[i]] <- aggr_counts[, column_idx]
  names(counts_ls)[i] <- cluster_names[i]
}

str(counts_ls)
```

```{r metadata df}
metadata <- colData(sce) %>%
  as.data.frame() %>%
  dplyr::select(genotype, sex, sample, Ever_Learner, Sac_Learner)

metadata <- metadata[!duplicated(metadata), ]
metadata

rownames(metadata) <- metadata$sample
```

```{r check columns}
t <- table(colData(sce)$sample,
           colData(sce)$cell_type)
t[1:9, 1:6]
```

```{r save table cells per geno}
write.csv(t, "/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/250301_tableofcellspergenotype.csv",
          quote = FALSE,
          row.names = TRUE)
```

```{r check clusters and sample ids}
for (i in 1:length(counts_ls)) {
    data <- data.frame(cluster_sample_id = colnames(counts_ls[[i]]))
    print(data)
}
```

```{r list metadata per cluster}
metadata_ls <- list()

for (i in 1:length(counts_ls)) {
  print(names(counts_ls)[i])
  data <- data.frame(cluster_sample_id = colnames(counts_ls[[i]]))
  data$cluster_id <- tstrsplit(data$cluster_sample_id, "_")[[1]]
  data$sample <- str_extract(data$cluster_sample_id,  "(?<=_).*")
  
  idx <- which(colnames(t) == unique(data$cluster_id))
  cell_counts <- t[, idx]
  
  cell_counts <- cell_counts[cell_counts > 0]
  
  sample_order <- match(data$sample, names(cell_counts))
  cell_counts <- cell_counts[sample_order]
  
  data$cell_count <- cell_counts
  
  data <- plyr::join(data, metadata,
                     by = intersect(names(data), names(metadata)))
  
  rownames(data) <- data$cluster_sample_id
  
  metadata_ls[[i]] <- data
  metadata_ls[[i]]$genotype <- factor(metadata_ls[[i]]$genotype, levels = c("WT", "Het"))
  metadata_ls[[i]]$sex <- factor(metadata_ls[[i]]$sex, levels = c("M", "F"))
  metadata_ls[[i]]$dosage <- as.numeric(metadata_ls[[i]]$genotype)
  names(metadata_ls)[i] <- unique(data$cluster_id)
  metadata_ls[[i]]$Sac_Learner <- factor(metadata_ls[[i]]$Sac_Learner, levels = c("Learner", "Non_Learner"))
  metadata_ls[[i]]$Ever_Learner <- factor(metadata_ls[[i]]$Ever_Learner, levels = c("1", "0"))
}

str(metadata_ls)
```

# Analyze with DESeq2

```{r check names}
names(counts_ls)

all(names(counts_ls) == names(metadata_ls))
```


```{r loop thru DESeq2 for geno}
for (name in cluster_names){
  idx <- which(names(counts_ls) == name)
  cluster_counts <- counts_ls[[idx]]
  cluster_metadata <- metadata_ls[[idx]]
  
  all(colnames(cluster_counts) == rownames(cluster_metadata))
  
  print(name)
  print(idx)

dds <- DESeqDataSetFromMatrix(round(cluster_counts),
                            colData = cluster_metadata,
                           design = ~ genotype)

 
  dds <- DESeq(dds)
  
  res <- results(dds,
               name = 'genotype_Het_vs_WT',
              alpha = 0.05)

  
  res_tbl <- res %>%
    data.frame() %>%
    rownames_to_column(var = 'gene') %>%
    as_tibble() %>%
    arrange(padj)
  
  padj_cutoff <- 0.1
  
  sig_res <- dplyr::filter(res_tbl, padj < padj_cutoff) %>%
    dplyr::arrange(padj)
  
  sig_res
  
  name.cleaned <- gsub('/', '_', name)
  write.csv(sig_res, paste0("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Geno/", name.cleaned, " signif genes.csv"),
            quote = FALSE,
            row.names = TRUE)
}
```

```{r loop thru DESeq2 for sex}
for (name in cluster_names){
  idx <- which(names(counts_ls) == name)
  cluster_counts <- counts_ls[[idx]]
  cluster_metadata <- metadata_ls[[idx]]
  

  all(colnames(cluster_counts) == rownames(cluster_metadata))
  
  print(name)
  print(idx)

  dds <- DESeqDataSetFromMatrix(round(cluster_counts),
                                colData = cluster_metadata,
                                design = ~ sex)

  dds <- DESeq(dds)
 
  res <- results(dds,
                 name = 'sex_F_vs_M',
                 alpha = 0.05)


  res_tbl <- res %>%
    data.frame() %>%
    rownames_to_column(var = 'gene') %>%
    as_tibble() %>%
    arrange(padj)
  
  padj_cutoff <- 0.1
  
  sig_res <- dplyr::filter(res_tbl, padj < padj_cutoff) %>%
    dplyr::arrange(padj)
  
  sig_res
  
  name.cleaned <- gsub('/', '_', name)
  write.csv(sig_res, paste0("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Sex/", name.cleaned, " signif genes.csv"),
            quote = FALSE,
            row.names = TRUE)
}
```

```{r loop thru DESeq2 for learner}
for (name in cluster_names){
  idx <- which(names(counts_ls) == name)
  cluster_counts <- counts_ls[[idx]]
  cluster_metadata <- metadata_ls[[idx]]
  

  all(colnames(cluster_counts) == rownames(cluster_metadata))
  
  print(name)
  print(idx)

     dds <- DESeqDataSetFromMatrix(round(cluster_counts),
                                 colData = cluster_metadata,
                                 design = ~ Sac_Learner)

  dds <- DESeq(dds)
 
    res <- results(dds,
                 name = 'Sac_Learner_Non_Learner_vs_Learner',
                 alpha = 0.05)

  res_tbl <- res %>%
    data.frame() %>%
    rownames_to_column(var = 'gene') %>%
    as_tibble() %>%
    arrange(padj)
  
  padj_cutoff <- 0.1
  
  sig_res <- dplyr::filter(res_tbl, padj < padj_cutoff) %>%
    dplyr::arrange(padj)
  
  sig_res
  
  name.cleaned <- gsub('/', '_', name)
  write.csv(sig_res, paste0("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Learner/", name.cleaned, " signif genes.csv"),
            quote = FALSE,
            row.names = TRUE)
}
```

# Summary tables and figures

## By Geno

### Bar plot
```{r DEG results df for geno}
file_list <- list.files("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Geno", pattern = "signif genes.csv", full.names = TRUE)

results_df <- data.frame(file = character(), 
                         wt_enrich = integer(), 
                         mut_enrich = integer(), 
                         stringsAsFactors = FALSE)

for (file in file_list) {
  file_df <- read.csv(file)
  wt_enrich <- sum(file_df$log2FoldChange < 0)
  mut_enrich <- sum(file_df$log2FoldChange > 0)
  
  results_df <- rbind(results_df, 
                      data.frame(file = file, 
                                 wt_enrich = wt_enrich, 
                                 mut_enrich = mut_enrich, 
                                 stringsAsFactors = FALSE))
}

results_df$file <- sub("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Geno/", "", results_df$file)
results_df$file <- sub(" signif genes.csv", "", results_df$file)

write.csv(results_df, file = "/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Geno/DEG table.csv", sep = ",")
```

```{r plot DEG results geno}
results_df$cluster <- results_df$file
results_df$cluster <- factor(results_df$cluster, levels = rev(unique(results_df$cluster)))

# First, make F_enrich negative to flip it to the left side
results_df$wt_enrich_neg <- -results_df$wt_enrich

# Create the combined plot
comb.plot <- ggplot(results_df) +
  # Male enrichment (right side)
  geom_bar(aes(x = wt_enrich_neg, y = cluster, fill = "WT"),
           stat = "identity",
           linewidth = 0.2) +
  # Female enrichment (left side)
  geom_bar(aes(x = mut_enrich, y = cluster, fill = "Het"),
           stat = "identity",
           linewidth = 0.2) +
  # Set the axis limits symmetrically
  scale_x_continuous(
    limits = c(-60, 30),
    breaks = seq(-60, 30, 10),
    labels = abs(seq(-60, 30, 10))  # Use absolute values for axis labels
  ) +
  scale_fill_manual(values = palette_geno) +
  labs(x = "Number of DEGs", 
       y = "Cluster",
       fill = "Genotype") +  # Add title for the legend
  theme_classic() +
  theme(        axis.text.y = element_text(color = rep(c("#000000", "#555555"), 
                                             length.out = length(unique(results_df$cluster))))) +
  geom_vline(xintercept = 0, linewidth = 0.2)

comb.plot
ggsave('/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/intermed-figs/250301_DEGsbyGenoall.pdf',
       device = pdf,
       width = 6,
       height = 6)
```

```{r filter results for sig clusters}
# Filter the data frame for P.Value < 0.051
sig_results_df <- results_df[((results_df$wt_enrich != 0) + (results_df$mut_enrich != 0)) >1, ]

# Extract the BaselineProp.clusters for the significant results
significant_cluster_names <-sort( as.character(sig_results_df$file))

# Print the list of significant clusters
print(significant_cluster_names)
```

```{r plot sig DEG results for geno}
# Create the combined plot
sig.comb.plot <- ggplot(sig_results_df) +
  # Male enrichment (right side)
  geom_bar(aes(x = wt_enrich_neg, y = cluster, fill = "WT"),
           stat = "identity",
           linewidth = 0.2) +
  # Female enrichment (left side)
  geom_bar(aes(x = mut_enrich, y = cluster, fill = "Het"),
           stat = "identity",
           linewidth = 0.2) +
  # Set the axis limits symmetrically
  scale_x_continuous(
    limits = c(-60, 30),
    breaks = seq(-60, 30, 10),
    labels = abs(seq(-60, 30, 10))  # Use absolute values for axis labels
  ) +
  scale_fill_manual(values = palette_geno) +
  labs(x = "Number of DEGs", 
       y = "",
       fill = "Genotype") +  # Add title for the legend
  theme_classic() +
  theme(axis.text.y = element_blank(),
        legend.position = "none") +
  geom_vline(xintercept = 0, linewidth = 0.2)

sig.comb.plot

 ggsave(filename = "/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/intermed-figs/250301_DEGsbygeno_sig.pdf",
        width = 8,
        height = 12,
        dpi = 300)
```

### Summary tables

```{r DEG table with all output geno}
file_list <- list.files("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Geno", pattern = "signif genes.csv", full.names = TRUE)
file_list <- sub("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Geno/", "", results_df$file)

concatenated_data <- data.frame()

for (file in file_list){
  current_data <- read.csv(paste0("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Geno/", file, " signif genes.csv"), sep = ',', header = TRUE)
  if(dim(current_data)[1] > 0) {
  current_data$cellType <- sub("signif genes.csv", "", file)
  concatenated_data <- rbind(concatenated_data, current_data)
  }
}

write.csv(concatenated_data, file = '/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Geno/DEGs_table.csv')
```

List of unique genes
```{r unique DEG list geno}
uniqueDEGs <- as.data.frame(unique(concatenated_data$gene))
print(uniqueDEGs, sep = ",")

write_csv(uniqueDEGs, file = "/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Geno/uniqueDEGs.csv")
```

### Volcano plot

```{r results table for plotting geno}
res_all <- concatenated_data %>%
  mutate(
    avg_log2FC = log2FoldChange,
    p_val_adj = padj,
    cell_type = cellType,
    change = case_when(
      p_val_adj >= 0.05 ~ "None",  
      avg_log2FC > 0 ~ "Up",       
      avg_log2FC < 0 ~ "Down",    
      TRUE ~ "None"                 
    ),
    # Force the factor levels order
    change = factor(change, levels = c("Up", "Down", "None"))
  )
```

```{r filter res for cluster of interest geno}
# Filter for specific cluster
res_table = res_all %>% filter(cell_type == "TH-Sema5b-Ntrk-Glut")

# Get top 10 most significant genes
top_genes = res_table %>% 
  filter(change != 'None') %>%
  slice_min(p_val_adj, n = 10) %>%
  pull(gene)

x_range = max(abs(res_table$avg_log2FC), na.rm = T) * 1.05
y_range = max(-log10(res_table$p_val_adj), na.rm = T) * 1.1
```

```{r volcano plot for cluster of interest geno}
p <- ggplot(res_table, aes(x = avg_log2FC, y = -log10(p_val_adj))) +
  geom_point(aes(color = change), size = 2) +
  geom_text_repel(
    data = res_table %>% filter(gene %in% top_genes),
    aes(label = gene),
    size = 3,
    max.overlaps = Inf,
    box.padding = 0.5,
    segment.size = 0.5,
    segment.color = "grey50"
  ) +
  ggtitle('DEGs (TH-Sema5b-Ntrk-Glut)') +
  xlab(expression(log[2]("Fold Change"))) +
  ylab(expression(-log[10]("adjusted p-value"))) +
  geom_hline(yintercept = -log10(0.05), colour = "grey40", size = 0.7) +
  geom_vline(xintercept = 0, colour = "black") +
  ggplot2::xlim(-x_range, x_range) +
  ggplot2::ylim(0, y_range) +
  scale_y_continuous(expand = c(0, 0.1)) +  # Back to regular scaling
  theme_classic() +
  scale_color_manual(
    values = c(Up = '#b80f0a', Down = '#0a6fb8', None = 'grey'),
    breaks = c("Up", "Down", "None"),
    labels = c('Up-regulated', 'Down-regulated', 'Not significant')
  ) +
  theme(
    axis.text.x = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.title = element_text(size = 14),
    legend.title = element_blank(),
    legend.text = element_text(size = 12),
    axis.line = element_line(size = 0.8),
    plot.title = element_text(size = 16)
  )

p

ggsave(filename = "/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/250301_TH-Sema5b-Ntrk-Glut_geno_volcano.pdf",
       width = 6,
       height = 4,
       device = pdf)
```


### GO
```{r GO dot plot for cluster of interest}
# Specify cluster name
name <- "TH-Sema5b-Ntrk-Glut"

# GO Analysis
go_table <- read.csv(paste0("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Geno/", name, " signif genes.csv"), sep = ',',
                     header = TRUE)

# Separate genes by direction of change - explicitly get gene names as character vector
wt_enrich <- as.character(go_table$gene[go_table$log2FoldChange > 0])
mut_enrich <- as.character(go_table$gene[go_table$log2FoldChange < 0])


# Convert gene symbols to ENTREZ IDs
geneid.ls <- AnnotationDbi::select(org.Mm.eg.db, 
                    keys = wt_enrich, 
                    columns=c("ENTREZID", "SYMBOL"), 
                    keytype = "SYMBOL")
wt.gene.s <- geneid.ls$ENTREZID
wt.gene.s <- wt.gene.s[which(!is.na(wt.gene.s))]
wt.gene.s <- unique(wt.gene.s)

geneid.ls <- AnnotationDbi::select(org.Mm.eg.db, 
                    keys = mut_enrich, 
                    columns=c("ENTREZID", "SYMBOL"), 
                    keytype = "SYMBOL")
mut.gene.s <- geneid.ls$ENTREZID
mut.gene.s <- mut.gene.s[which(!is.na(mut.gene.s))]
mut.gene.s <- unique(mut.gene.s)

gene_list <- list(Mutant=mut.gene.s, Wildtype=wt.gene.s)

# Run GO enrichment comparison
compGO <- compareCluster(geneCluster = gene_list,
                        fun = 'enrichGO',
                        pvalueCutoff = 0.05,
                        pAdjustMethod = 'BH',
                        OrgDb = org.Mm.eg.db,
                        ont = 'BP')

# Create dotplot visualization
dotplot(compGO, showCategory = 4, title = 'GO Enrichment Analysis') +
  theme(text = element_text(size = 10),
        axis.text.y = element_text(size=10))

ggsave(paste0("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Geno/", name, " GO enrichment dotplot.pdf"),
       device = pdf,
       width = 5,
       height = 5)
```

```{r GO combined plot for all sig clusters geno}
# GO Analysis - Combined plot
# First get a background list of all genes
all.genes <- as.character(rownames(WT.only@assays$RNA))
geneid.ls <- AnnotationDbi::select(org.Mm.eg.db, 
                    keys = all.genes, 
                    columns=c("ENTREZID", "SYMBOL"), 
                    keytype = "SYMBOL")
alwt.gene.s <- geneid.ls$ENTREZID
alwt.gene.s <- alwt.gene.s[which(!is.na(alwt.gene.s))]
alwt.gene.s <- unique(alwt.gene.s)

# Initialize lists to store genes by cluster
cluster_genes_list <- list()

# Process each cluster
for (name in significant_cluster_names) {
  print(paste("Processing cluster:", name))
  
  # Read DEG results
  go_table <- read.csv(paste0("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Geno/", name, " signif genes.csv"), sep = ',',
                       header = TRUE)
  
  # Get significant genes
  sig_genes <- as.character(go_table$gene)
  
  # Convert to ENTREZ IDs
  geneid.ls <- AnnotationDbi::select(org.Mm.eg.db, 
                      keys = sig_genes, 
                      columns = c("ENTREZID", "SYMBOL"), 
                      keytype = "SYMBOL")
  gene.ls <- geneid.ls$ENTREZID
  gene.ls <- gene.ls[which(!is.na(gene.ls))]
  gene.ls <- unique(gene.ls)
  
  # Store in list
  cluster_genes_list[[name]] <- gene.ls
}

# Run GO enrichment comparison
compGO <- compareCluster(geneCluster = cluster_genes_list,
                        fun = 'enrichGO',
                        pvalueCutoff = 0.05,
                        pAdjustMethod = 'BH',
                        OrgDb = org.Mm.eg.db,
                        ont = 'BP',
                        universe = alwt.gene.s)

# Create dotplot visualization
p <- dotplot(compGO, showCategory = 10, 
            title = 'GO Enrichment Analysis by Cluster') +
  theme(text = element_text(size = 10),
        axis.text.y = element_text(size = 10),
        axis.text.x = element_text(angle = 45, hjust = 1),
        plot.margin = unit(c(1,1,1,1), "cm")) +
  scale_color_viridis_c(direction = -1)

p
# Save plot
ggsave("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Geno/cluster_GO_enrichment_dotplot.pdf",
       plot = p,
       width = 12,
       height = 10)
```


## By Sex

### Bar plot
```{r DEG results df for sex}
file_list <- list.files("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Sex/", pattern = "signif genes.csv", full.names = TRUE)

results_df <- data.frame(file = character(), 
                         F_enrich = integer(), 
                         M_enrich = integer(), 
                         stringsAsFactors = FALSE)

for (file in file_list) {
  file_df <- read.csv(file)
  F_enrich <- sum(file_df$log2FoldChange < 0)
  M_enrich <- sum(file_df$log2FoldChange > 0)
  
  results_df <- rbind(results_df, 
                      data.frame(file = file, 
                                 F_enrich = F_enrich, 
                                 M_enrich = M_enrich, 
                                 stringsAsFactors = FALSE))
}

results_df$file <- sub("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Sex//", "", results_df$file)
results_df$file <- sub(" signif genes.csv", "", results_df$file)

write.csv(results_df, file = "/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/DEG table.csv", sep = ",")
```

```{r plot DEG results sex}
results_df$cluster <- results_df$file
results_df$cluster <- factor(results_df$cluster, levels = rev(unique(results_df$cluster)))

# First, make F_enrich negative to flip it to the left side
results_df$F_enrich_neg <- -results_df$F_enrich

# Create the combined plot
comb.plot <- ggplot(results_df) +
  # Male enrichment (right side)
  geom_bar(aes(x = M_enrich, y = cluster, fill = "M"),
           stat = "identity",
           linewidth = 0.2) +
  # Female enrichment (left side)
  geom_bar(aes(x = F_enrich_neg, y = cluster, fill = "F"),
           stat = "identity",
           linewidth = 0.2) +
  # Set the axis limits symmetrically
  scale_x_continuous(
    limits = c(-30, 30),
    breaks = seq(-30, 30, 5),
    labels = abs(seq(-30, 30, 5))  # Use absolute values for axis labels
  ) +
  scale_fill_manual(values = palette_sex) +
  labs(x = "Number of DEGs", 
       y = "Cluster",
       fill = "Sex") +  # Add title for the legend
  theme_classic() +
  theme(        axis.text.y = element_text(color = rep(c("#000000", "#555555"), 
                                             length.out = length(unique(results_df$cluster))))) +
  geom_vline(xintercept = 0, linewidth = 0.2)

comb.plot
ggsave('/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/intermed-figs/250301_DEGsbySexall.pdf',
       device = pdf,
       width = 6,
       height = 6)
```

```{r plot sig DEG results for sex}
# Get sig clusters from proportions script
sig_cluster_names <- c("ARC-Mixed-Glut",
  "COAa-PAA-MEA-Barhl2-Glut",
  "HY-NSCs-1",
  "MEA-Glut",
  "MEA-BST-Mixed-Gaba",
  "PH-LHA-PVH-SO-PVa-Nxph4-Glut",
  "PVN-Mixed-Glut")

sig_results_df <- results_df %>%
  dplyr::filter(cluster %in% sig_cluster_names) %>%
  arrange(match(cluster, sig_cluster_names))


sig_results_df$cluster <- factor(sig_results_df$cluster, levels = rev(sig_cluster_names))


sig_plot <- 
# Create the combined plot
ggplot(sig_results_df) +
  # Male enrichment (right side)
  geom_bar(aes(x = M_enrich, y = cluster, fill = "M"),
           stat = "identity",
           linewidth = 0.2) +
  # Female enrichment (left side)
  geom_bar(aes(x = F_enrich_neg, y = cluster, fill = "F"),
           stat = "identity",
           linewidth = 0.2) +
  # Set the axis limits symmetrically
  scale_x_continuous(
    limits = c(-30, 30),
    breaks = seq(-30, 30, 5),
    labels = abs(seq(-30, 30, 5))  # Use absolute values for axis labels
  ) +
  scale_fill_manual(values = palette_sex) +
  labs(x = "Number of DEGs", 
       y = "",
       fill = "Sex") +  # Add title for the legend
  theme_classic() +
  theme(axis.text.y = element_blank()) +
  geom_vline(xintercept = 0, linewidth = 0.2)
  
sig_plot

 ggsave(filename = "/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/intermed-figs/250219_DEGsbysex_sig.pdf",
        width = 8,
        height = 12,
        dpi = 300)
```


### Summary tables

```{r DEG table with all output sex}
file_list <- list.files("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Sex", pattern = "signif genes.csv", full.names = TRUE)
file_list <- sub("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Sex/", "", results_df$file)


concatenated_data <- data.frame()

for (file in file_list){
  current_data <- read.csv(paste0("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Sex/", file, " signif genes.csv"), sep = ',', header = TRUE)
  if(dim(current_data)[1] > 0) {
  current_data$cellType <- sub("signif genes.csv", "", file)
  concatenated_data <- rbind(concatenated_data, current_data)
  }
}

write.csv(concatenated_data, file = '/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Sex/250301_DEGs_table.csv')
```

```{r unique DEGs list sex}
uniqueDEGs <- as.data.frame(unique(concatenated_data$gene))
print(uniqueDEGs, sep = ",")

write_csv(uniqueDEGs, file = "/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Sex/250301_uniqueDEGs.csv")
```

### Volcano plots

```{r results table for plotting sex}
res_all <- concatenated_data %>%
  mutate(
    avg_log2FC = log2FoldChange,
    p_val_adj = padj,
    cell_type = cellType,
    change = case_when(
      p_val_adj >= 0.05 ~ "None",  
      avg_log2FC > 0 ~ "Up",       
      avg_log2FC < 0 ~ "Down",    
      TRUE ~ "None"                 
    ),
    # Force the factor levels order
    change = factor(change, levels = c("Up", "Down", "None"))
  )
```

```{r filter res for cluster of interest sex}
# Filter for specific cluster
res_table = res_all %>% filter(cell_type == "PO-MEA-BST-Mixed-Gaba")

# Get top 10 most significant genes
top_genes = res_table %>% 
  filter(change != 'None') %>%
  slice_min(p_val_adj, n = 10) %>%
  pull(gene)

x_range = max(abs(res_table$avg_log2FC), na.rm = T) * 1.05
y_range = max(-log10(res_table$p_val_adj), na.rm = T) * 1.1
```

```{r volcano plot for cluster of interest sex}
p <- ggplot(res_table, aes(x = avg_log2FC, y = -log10(p_val_adj))) +
  geom_point(aes(color = change), size = 2) +
  geom_text_repel(
    data = res_table %>% filter(gene %in% top_genes),
    aes(label = gene),
    size = 3,
    max.overlaps = Inf,
    box.padding = 0.5,
    segment.size = 0.5,
    segment.color = "grey50"
  ) +
  ggtitle('DEGs (MEA-BST-Mixed-Gaba)') +
  xlab(expression(log[2]("Fold Change"))) +
  ylab(expression(-log[10]("adjusted p-value"))) +
  geom_hline(yintercept = -log10(0.05), colour = "grey40", size = 0.7) +
  geom_vline(xintercept = 0, colour = "black") +
  ggplot2::xlim(-x_range, x_range) +
  ggplot2::ylim(0, y_range) +
  scale_y_continuous(expand = c(0, 0.1)) +  # Back to regular scaling
  theme_classic() +
  scale_color_manual(
    values = c(Up = '#b80f0a', Down = '#0a6fb8', None = 'grey'),
    breaks = c("Up", "Down", "None"),
    labels = c('Up-regulated', 'Down-regulated', 'Not significant')
  ) +
  theme(
    axis.text.x = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.title = element_text(size = 14),
    legend.title = element_blank(),
    legend.text = element_text(size = 12),
    axis.line = element_line(size = 0.8),
    plot.title = element_text(size = 16)
  )

p

ggsave(filename = "/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/250301_PO-MEA-BST-Mixed-Gaba_volcano.pdf", 
       width = 6,
       height = 4,
       device = pdf)
```

Now let's remove X and Y chromosome genes.
```{r filter out X and Y chr genes}
xy_genes <- c(
  # X chromosome genes
  "Xist", "Tsx", "Jpx", "Ftx", "Kdm6a", "Kdm5c", "Ddx3x", "Usp9x", "Uba1", "Zfx",
  # Y chromosome genes
  "Uty", "Kdm5d", "Eif2s3y", "Ddx3y", "Usp9y", "Zfy1", "Zfy2", "Sry", "Rbmy"
)

# Filter the data frame to remove X and Y genes
res_filtered <- res_all %>%
  filter(!gene %in% xy_genes)
```

```{r re-filter for cluster of interest sex}
# Filter for specific cluster
res_table_filtered = res_filtered %>% filter(cell_type == "PO-MEA-BST-Mixed-Gaba")

# Get top 10 most significant genes
top_genes = res_table_filtered %>% 
  filter(change != 'None') %>%
  slice_min(p_val_adj, n = 10) %>%
  pull(gene)

x_range = max(abs(res_table_filtered$avg_log2FC), na.rm = T) * 1.05
y_range = max(-log10(res_table_filtered$p_val_adj), na.rm = T) * 1.1
```

```{r volcano plot for cluster of interest sex without X/Y chr DEGs}
p <- ggplot(res_table_filtered, aes(x = avg_log2FC, y = -log10(p_val_adj))) +
  geom_point(aes(color = change), size = 2) +
  geom_text_repel(
    data = res_table_filtered %>% filter(gene %in% top_genes),
    aes(label = gene),
    size = 3,
    max.overlaps = Inf,
    box.padding = 0.5,
    segment.size = 0.5,
    segment.color = "grey50"
  ) +
  ggtitle('DEGs (MEA-BST-Mixed-Gaba)') +
  xlab(expression(log[2]("Fold Change"))) +
  ylab(expression(-log[10]("adjusted p-value"))) +
  geom_hline(yintercept = -log10(0.05), colour = "grey40", size = 0.7) +
  geom_vline(xintercept = 0, colour = "black") +
  ggplot2::xlim(-x_range, x_range) +
  ggplot2::ylim(0, y_range) +
  scale_y_continuous(expand = c(0, 0.1)) +  # Back to regular scaling
  theme_classic() +
  scale_color_manual(
    values = c(Up = '#b80f0a', Down = '#0a6fb8', None = 'grey'),
    breaks = c("Up", "Down", "None"),
    labels = c('Up-regulated', 'Down-regulated', 'Not significant')
  ) +
  theme(
    axis.text.x = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.title = element_text(size = 14),
    legend.title = element_blank(),
    legend.text = element_text(size = 12),
    axis.line = element_line(size = 0.8),
    plot.title = element_text(size = 16)
  )

p

ggsave(filename = "/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/250301_PO-MEA-BST-Mixed-Gaba_volcano_noXYgenes.pdf", 
       width = 6,
       height = 4,
       device = pdf)
```



## By Learner

### Bar plot
```{r DEG results df for sac learner}
file_list <- list.files("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Learner", pattern = "signif genes.csv", full.names = TRUE)

results_df <- data.frame(file = character(), 
                         L_enrich = integer(), 
                         NL_enrich = integer(), 
                         stringsAsFactors = FALSE)

for (file in file_list) {
  file_df <- read.csv(file)
  L_enrich <- sum(file_df$log2FoldChange < 0)
  NL_enrich <- sum(file_df$log2FoldChange > 0)
  
  results_df <- rbind(results_df, 
                      data.frame(file = file, 
                                 L_enrich = L_enrich, 
                                 NL_enrich = NL_enrich, 
                                 stringsAsFactors = FALSE))
}

results_df$file <- sub("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Learner/", "", results_df$file)
results_df$file <- sub(" signif genes.csv", "", results_df$file)

write.csv(results_df, file = "/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Learner/DEG table.csv", sep = ",")
```

```{r plot DEG results sac learner}
rownames(results_df) <- results_df$file
results_df$cluster <- factor(results_df$file)
results_df <- drop_na(results_df)

# First, make NL_enrich negative to flip it to the left side
results_df$NL_enrich_neg <- -results_df$NL_enrich

# Create the combined plot
comb.plot <- ggplot(results_df) +
  # Male enrichment (right side)
  geom_bar(aes(x = L_enrich, y = cluster, fill = "Learner"),
           stat = "identity",
           linewidth = 0.2) +
  # Female enrichment (left side)
  geom_bar(aes(x = NL_enrich_neg, y = cluster, fill = "Non_Learner"),
           stat = "identity",
           linewidth = 0.2) +
  # Set the axis limits symmetrically
  scale_x_continuous(
    limits = c(-30, 30),
    breaks = seq(-30, 30, 5),
    labels = abs(seq(-30, 30, 5))  # Use absolute values for axis labels
  ) +
  scale_fill_manual(values = palette_learner) +
  labs(x = "Number of DEGs", 
       y = "Cluster",
       fill = "Sac_Learner") +  # Add title for the legend
  theme_classic() +
  theme(axis.text.y = element_text(color = rep(c("#000000", "#555555"), 
                      length.out = length(unique(results_df$cluster))))) +
  geom_vline(xintercept = 0, linewidth = 0.2)

comb.plot
ggsave('/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/intermed-figs/250301_DEGsbyLearnerall.pdf',
       device = pdf,
       width = 6,
       height = 6)
```


```{r filter results for sig clusters}
# Filter the data frame for P.Value < 0.051
sig_results_df <- results_df[((results_df$NL_enrich != 0) + (results_df$L_enrich != 0)) >1, ]

# Extract the BaselineProp.clusters for the significant results
significant_cluster_names <-sort(rev(as.character(sig_results_df$file)))

# Print the list of significant clusters
print(significant_cluster_names)


sig_results_df$cluster <- factor(sig_results_df$cluster, 
                                levels = rev(significant_cluster_names))
```

```{r plot sig DEG results for sac learner}
# Create the combined plot
sig.comb.plot <- ggplot(sig_results_df) +
  # Male enrichment (right side)
  geom_bar(aes(x = L_enrich, y = cluster, fill = "Learner"),
           stat = "identity",
           linewidth = 0.2) +
  # Female enrichment (left side)
  geom_bar(aes(x = NL_enrich_neg, y = cluster, fill = "Non_Learner"),
           stat = "identity",
           linewidth = 0.2) +
  # Set the axis limits symmetrically
  scale_x_continuous(
    limits = c(-30, 30),
    breaks = seq(-30, 30, 5),
    labels = abs(seq(-30, 30, 5))  # Use absolute values for axis labels
  ) +
  scale_fill_manual(values = palette_learner) +
  labs(x = "Number of DEGs", 
       y = "Cluster",
       fill = "Sac_Learner") +  # Add title for the legend
  theme_classic() +
  theme(axis.text.y = element_text(color = rep(c("#000000", "#555555"), 
                      length.out = length(unique(results_df$cluster))))) +
  geom_vline(xintercept = 0, linewidth = 0.2)

sig.comb.plot


 ggsave(filename = "/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/250301_DEGsbylearner.pdf",
        width = 8,
        height = 12,
        dpi = 300)
```

### Summary tables
```{r DEG table with all output sac learner}
file_list <- sub("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Learner/", "", results_df$file)

concatenated_data <- data.frame()

for (file in file_list){
  current_data <- read.csv(paste0("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Learner/", file, " signif genes.csv"), sep = ',', header = TRUE)
  if(dim(current_data)[1] > 0) {
  current_data$cellType <- sub("signif genes.csv", "", file)
  concatenated_data <- rbind(concatenated_data, current_data)
  }
}

write.csv(concatenated_data, file = '/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Learner/250301-DEGs_table.csv')
```

```{r unique DEGs list sac learner}
uniqueDEGs <- as.data.frame(unique(concatenated_data$gene))
print(uniqueDEGs, sep = ",")

write_csv(uniqueDEGs, file = "/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Learner/250301_uniqueDEGs.csv")
```

### Volcano plot
```{r results table for plotting geno}
res_all <- concatenated_data %>%
  mutate(
    avg_log2FC = log2FoldChange,
    p_val_adj = padj,
    cell_type = cellType,
    change = case_when(
      p_val_adj >= 0.05 ~ "None",  
      avg_log2FC > 0 ~ "Up",       
      avg_log2FC < 0 ~ "Down",    
      TRUE ~ "None"                 
    ),
    # Force the factor levels order
    change = factor(change, levels = c("Up", "Down", "None"))
  )
```


```{r filter res for cluster of interest geno}
# Filter for specific cluster
res_table = res_all %>% filter(cell_type == "PH-LHA-PVH-SO-PVa-Nxph4-Glut")

# Get top 10 most significant genes
top_genes = res_table %>% 
  filter(change != 'None') %>%
  slice_min(p_val_adj, n = 10) %>%
  pull(gene)

x_range = max(abs(res_table$avg_log2FC), na.rm = T) * 1.05
y_range = max(-log10(res_table$p_val_adj), na.rm = T) * 1.1
```


```{r volcano plot for cluster of interest geno}
p <- ggplot(res_table, aes(x = avg_log2FC, y = -log10(p_val_adj))) +
  geom_point(aes(color = change), size = 2) +
  ggrepel::geom_text_repel(
    data = res_table %>% filter(gene %in% top_genes),
    aes(label = gene),
    size = 3,
    max.overlaps = Inf,
    box.padding = 0.5,
    segment.size = 0.5,
    segment.color = "grey50"
  ) +
  ggtitle('DEGs (PH-LHA-PVH-SO-PVa-Nxph4-Glut)') +
  xlab(expression(log[2]("Fold Change"))) +
  ylab(expression(-log[10]("adjusted p-value"))) +
  geom_hline(yintercept = -log10(0.05), colour = "grey40", size = 0.7) +
  geom_vline(xintercept = 0, colour = "black") +
  ggplot2::xlim(-x_range, x_range) +
  ggplot2::ylim(0, y_range) +
  scale_y_continuous(expand = c(0, 0.1)) +  # Back to regular scaling
  theme_classic() +
  scale_color_manual(
    values = c(Up = '#b80f0a', Down = '#0a6fb8', None = 'grey'),
    breaks = c("Up", "Down", "None"),
    labels = c('Up-regulated', 'Down-regulated', 'Not significant')
  ) +
  theme(
    axis.text.x = element_text(size = 12),
    axis.text.y = element_text(size = 12),
    axis.title = element_text(size = 14),
    legend.title = element_blank(),
    legend.text = element_text(size = 12),
    axis.line = element_line(size = 0.8),
    plot.title = element_text(size = 16)
  )

p

ggsave(filename = "/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/250301_PH-LHA-PVH-SO-PVa-Nxph4-Glut_learner_volcano.pdf",
       width = 6,
       height = 4,
       device = pdf)
```

### GO
```{r GO dot plot for cluster of interest}
# Specify cluster name
name <- "PH-LHA-PVH-SO-PVa-Nxph4-Glut"

# GO Analysis
go_table <- read.csv(paste0("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Learner/", name, " signif genes.csv"), sep = ',',
                     header = TRUE)

# Separate genes by direction of change - explicitly get gene names as character vector
L_enrich <- as.character(go_table$gene[go_table$log2FoldChange > 0])
NL_enrich <- as.character(go_table$gene[go_table$log2FoldChange < 0])


# Convert gene symbols to ENTREZ IDs
geneid.ls <- AnnotationDbi::select(org.Mm.eg.db, 
                    keys = L_enrich, 
                    columns=c("ENTREZID", "SYMBOL"), 
                    keytype = "SYMBOL")
L.gene.ls <- geneid.ls$ENTREZID
L.gene.ls <- L.gene.ls[which(!is.na(L.gene.ls))]
L.gene.ls <- unique(L.gene.ls)

geneid.ls <- AnnotationDbi::select(org.Mm.eg.db, 
                    keys = NL_enrich, 
                    columns=c("ENTREZID", "SYMBOL"), 
                    keytype = "SYMBOL")
NL.gene.ls <- geneid.ls$ENTREZID
NL.gene.ls <- NL.gene.ls[which(!is.na(NL.gene.ls))]
NL.gene.ls <- unique(NL.gene.ls)

gene_list <- list(Non_Learner=NL.gene.ls, Learner=L.gene.ls)

# Run GO enrichment comparison
compGO <- compareCluster(geneCluster = gene_list,
                        fun = 'enrichGO',
                        pvalueCutoff = 0.05,
                        pAdjustMethod = 'BH',
                        OrgDb = org.Mm.eg.db,
                        ont = 'BP')

# Create dotplot visualization
dotplot(compGO, showCategory = 10, title = 'GO Enrichment Analysis') +
  theme(text = element_text(size = 10),
        axis.text.y = element_text(size=10))

ggsave(paste0("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Learner/", name, " GO enrichment dotplot.pdf"),
       device = pdf,
       width = 5,
       height = 7)
```

```{r GO combined plot for all sig clusters}
# GO Analysis - Combined plot
# First get a background list of all genes
all.genes <- as.character(rownames(WT.only@assays$RNA))
geneid.ls <- AnnotationDbi::select(org.Mm.eg.db, 
                    keys = all.genes, 
                    columns=c("ENTREZID", "SYMBOL"), 
                    keytype = "SYMBOL")
all.gene.ls <- geneid.ls$ENTREZID
all.gene.ls <- all.gene.ls[which(!is.na(all.gene.ls))]
all.gene.ls <- unique(all.gene.ls)

# Initialize lists to store genes by cluster
cluster_genes_list <- list()

# Process each cluster
for (name in significant_cluster_names) {
  print(paste("Processing cluster:", name))
  
  # Read DEG results
  go_table <- read.csv(paste0("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Learner/", name, " signif genes.csv"), sep = ',',
                       header = TRUE)
  
  # Get significant genes
  sig_genes <- as.character(go_table$gene)
  
  # Convert to ENTREZ IDs
  geneid.ls <- AnnotationDbi::select(org.Mm.eg.db, 
                      keys = sig_genes, 
                      columns = c("ENTREZID", "SYMBOL"), 
                      keytype = "SYMBOL")
  gene.ls <- geneid.ls$ENTREZID
  gene.ls <- gene.ls[which(!is.na(gene.ls))]
  gene.ls <- unique(gene.ls)
  
  # Store in list
  cluster_genes_list[[name]] <- gene.ls
}

# Run GO enrichment comparison
compGO <- compareCluster(geneCluster = cluster_genes_list,
                        fun = 'enrichGO',
                        pvalueCutoff = 0.05,
                        pAdjustMethod = 'BH',
                        OrgDb = org.Mm.eg.db,
                        ont = 'BP',
                        universe = all.gene.ls)

# Create dotplot visualization
p <- dotplot(compGO, showCategory = 10, 
            title = 'GO Enrichment Analysis by Cluster') +
  theme(text = element_text(size = 10),
        axis.text.y = element_text(size = 10),
        axis.text.x = element_text(angle = 45, hjust = 1),
        plot.margin = unit(c(1,1,1,1), "cm")) +
  scale_color_viridis_c(direction = -1)

p
# Save plot
ggsave("/scratch/jdlab/simona/socop-hyp-snRNAseq/results/GEX/Learner/cluster_GO_enrichment_dotplot.pdf",
       plot = p,
       width = 12,
       height = 10)
```

```{r session info}
sessionInfo()
```

R version 4.2.2 (2022-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.2 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] stats4    parallel  grid      stats     graphics  grDevices
 [7] utils     datasets  methods   base     

other attached packages:
 [1] ggrepel_0.9.3               DESeq2_1.38.3              
 [3] png_0.1-8                   apeglm_1.20.0              
 [5] pheatmap_1.0.12             SingleCellExperiment_1.20.1
 [7] SummarizedExperiment_1.28.0 GenomicRanges_1.50.2       
 [9] GenomeInfoDb_1.34.9         MatrixGenerics_1.10.0      
[11] matrixStats_0.63.0          reshape2_1.4.4             
[13] edgeR_3.40.2                Matrix.utils_0.9.8         
[15] DOSE_3.24.2                 org.Mm.eg.db_3.16.0        
[17] AnnotationDbi_1.60.2        IRanges_2.32.0             
[19] S4Vectors_0.36.2            Biobase_2.58.0             
[21] BiocGenerics_0.44.0         enrichplot_1.18.4          
[23] clusterProfiler_4.6.2       ReactomePA_1.42.0          
[25] magrittr_2.0.3              lubridate_1.9.2            
[27] forcats_1.0.0               purrr_1.0.1                
[29] readr_2.1.4                 tibble_3.2.0               
[31] tidyverse_2.0.0             compositions_2.0-5         
[33] tidyr_1.3.0                 biomaRt_2.54.1             
[35] data.table_1.14.8           gt_0.8.0                   
[37] limma_3.54.2                speckle_0.99.7             
[39] ggtree_3.6.2                ggrastr_1.0.1              
[41] stringr_1.5.0               ComplexHeatmap_2.14.0      
[43] SeuratWrappers_0.2.0        SeuratDisk_0.0.0.9021      
[45] ggpubr_0.6.0                viridis_0.6.2              
[47] viridisLite_0.4.1           qs_0.25.5                  
[49] scCustomize_1.1.1           RColorBrewer_1.1-3         
[51] cowplot_1.1.1               ggplot2_3.4.1              
[53] Matrix_1.5-1                patchwork_1.1.2            
[55] SeuratObject_4.1.3          Seurat_4.3.0               
[57] dplyr_1.1.0                

loaded via a namespace (and not attached):
  [1] rsvd_1.0.5             ica_1.0-3             
  [3] foreach_1.5.2          lmtest_0.9-40         
  [5] crayon_1.5.2           MASS_7.3-58.1         
  [7] nlme_3.1-160           backports_1.4.1       
  [9] GOSemSim_2.24.0        rlang_1.1.3           
 [11] XVector_0.38.0         HDO.db_0.99.1         
 [13] ROCR_1.0-11            irlba_2.3.5.1         
 [15] filelock_1.0.2         stringfish_0.15.7     
 [17] BiocParallel_1.32.6    rjson_0.2.21          
 [19] bit64_4.0.5            glue_1.6.2            
 [21] sctransform_0.3.5      vipor_0.4.5           
 [23] spatstat.sparse_3.0-1  spatstat.geom_3.1-0   
 [25] tidyselect_1.2.0       fitdistrplus_1.1-8    
 [27] XML_3.99-0.13          zoo_1.8-11            
 [29] xtable_1.8-4           evaluate_0.20         
 [31] cli_3.6.2              zlibbioc_1.44.0       
 [33] rstudioapi_0.14        miniUI_0.1.1.1        
 [35] sp_1.6-0               fastmatch_1.1-3       
 [37] treeio_1.22.0          shiny_1.7.4           
 [39] xfun_0.37              clue_0.3-64           
 [41] gson_0.1.0             cluster_2.1.4         
 [43] tidygraph_1.2.3        KEGGREST_1.38.0       
 [45] ape_5.7-1              listenv_0.9.0         
 [47] Biostrings_2.66.0      future_1.32.0         
 [49] withr_2.5.0            bitops_1.0-7          
 [51] ggforce_0.4.1          plyr_1.8.8            
 [53] coda_0.19-4            pillar_1.8.1          
 [55] RcppParallel_5.1.7     GlobalOptions_0.1.2   
 [57] cachem_1.0.7           hdf5r_1.3.8           
 [59] GetoptLong_1.0.5       graphite_1.44.0       
 [61] paletteer_1.5.0        vctrs_0.5.2           
 [63] ellipsis_0.3.2         generics_0.1.3        
 [65] RApiSerialize_0.1.2    tools_4.2.2           
 [67] beeswarm_0.4.0         munsell_0.5.0         
 [69] tweenr_2.0.2           fgsea_1.24.0          
 [71] DelayedArray_0.24.0    fastmap_1.1.1         
 [73] compiler_4.2.2         abind_1.4-5           
 [75] httpuv_1.6.9           plotly_4.10.1         
 [77] GenomeInfoDbData_1.2.9 gridExtra_2.3         
 [79] lattice_0.20-45        deldir_1.0-6          
 [81] utf8_1.2.3             later_1.3.0           
 [83] BiocFileCache_2.6.1    jsonlite_1.8.4        
 [85] scales_1.2.1           graph_1.76.0          
 [87] tidytree_0.4.2         pbapply_1.7-0         
 [89] carData_3.0-5          lazyeval_0.2.2        
 [91] promises_1.2.0.1       car_3.1-1             
 [93] doParallel_1.0.17      goftest_1.2-3         
 [95] spatstat.utils_3.0-2   reticulate_1.28       
 [97] rmarkdown_2.20         textshaping_0.3.6     
 [99] statmod_1.5.0          Rtsne_0.16            
[101] downloader_0.4         uwot_0.1.14           
[103] igraph_1.5.0           numDeriv_2016.8-1.1   
[105] survival_3.4-0         yaml_2.3.7            
[107] systemfonts_1.0.4      htmltools_0.5.4       
[109] memoise_2.0.1          locfit_1.5-9.7        
[111] graphlayouts_0.8.4     digest_0.6.31         
[113] mime_0.12              rappdirs_0.3.3        
[115] emdbook_1.3.12         RSQLite_2.3.0         
[117] yulab.utils_0.0.6      future.apply_1.10.0   
[119] remotes_2.4.2          blob_1.2.3            
[121] ragg_1.2.5             splines_4.2.2         
[123] labeling_0.4.2         rematch2_2.1.2        
[125] Cairo_1.6-0            RCurl_1.98-1.10       
[127] broom_1.0.4            hms_1.1.2             
[129] colorspace_2.1-0       BiocManager_1.30.20   
[131] ggbeeswarm_0.7.1       shape_1.4.6           
[133] aplot_0.1.10           Rcpp_1.0.10           
[135] RANN_2.6.1             mvtnorm_1.1-3         
[137] circlize_0.4.15        fansi_1.0.4           
[139] tzdb_0.3.0             parallelly_1.34.0     
[141] R6_2.5.1               ggridges_0.5.4        
[143] lifecycle_1.0.3        curl_5.0.0            
[145] ggsignif_0.6.4         leiden_0.4.3          
[147] snakecase_0.11.0       robustbase_0.95-0     
[149] qvalue_2.30.0          RcppAnnoy_0.0.20      
[151] iterators_1.0.14       spatstat.explore_3.0-6
[153] htmlwidgets_1.6.1      polyclip_1.10-4       
[155] shadowtext_0.1.2       timechange_0.2.0      
[157] gridGraphics_0.5-1     reactome.db_1.82.0    
[159] globals_0.16.2         spatstat.random_3.1-4 
[161] tensorA_0.36.2         bdsmatrix_1.3-6       
[163] progressr_0.13.0       codetools_0.2-18      
[165] GO.db_3.16.0           prettyunits_1.1.1     
[167] dbplyr_2.3.1           gtable_0.3.1          
[169] DBI_1.1.3              bayesm_3.1-5          
[171] ggfun_0.0.9            tensor_1.5            
[173] httr_1.4.5             KernSmooth_2.23-20    
[175] vroom_1.6.1            stringi_1.7.12        
[177] progress_1.2.2         farver_2.1.1          
[179] annotate_1.76.0        xml2_1.3.3            
[181] bbmle_1.0.25           grr_0.9.5             
[183] geneplotter_1.76.0     ggplotify_0.1.0       
[185] scattermore_0.8        DEoptimR_1.0-11       
[187] bit_4.0.5              scatterpie_0.1.8      
[189] spatstat.data_3.0-1    ggraph_2.1.0          
[191] janitor_2.2.0          pkgconfig_2.0.3       
[193] ggprism_1.0.4          rstatix_0.7.2         
[195] knitr_1.42 
# End
